blaze/expr/reductions.py

```diff
@@ -99,6 +134,41 @@ def __str__(self):
         else:
             return '%s(%s)' % (name, self._child)
 
+    def over(self, kind='ROWS',
+             group_by=None, sort_by=None, preceding=None, following=None):
+        kind = kind.upper()
+        if kind != 'ROWS' and kind != 'RANGE':
+            raise ValueError(
+                '"kind" must be either "ROWS" or "RANGE" (case insensitive)'
+            )
+        if preceding is not None and not isinstance(
+            preceding, (numbers.Integral, np.integer)
+        ):
+            raise TypeError(
+                '"preceding" must be an integer, got %s of type %r' % (
+                    preceding, type(preceding).__name__
+                )
+            )
+        if following is not None and not isinstance(
+            following, (numbers.Integral, np.integer)
+        ):
+            raise TypeError(
+                '"following" must be an integer, got %s of type %r' % (
+                    following, type(following).__name__
+                )
+            )
+        return Window(
+            self,
+            kind=kind,
+            group_by=group_by,
+            sort_by=sort_by,
+            preceding=preceding,
+            following=following,
+            _asdshape=DataShape(
+                *(self._child._child.shape + (self._child.schema,))
```

Comment:
no, because when we make new leaves in `compute` from the results of computed expressions `self._child` will be a leaf by the time we get to the `compute_up` for `Window` and therefor `self._child._child` won't exist